packages/cli/src/lib/snapshot/snapshot.ts

```diff
@@ -0,0 +1,37 @@
+import {ResourceSnapshotsModel} from '@coveord/platform-client';
+import ResourceSnapshots from '@coveord/platform-client/dist/definitions/resources/ResourceSnapshots/ResourceSnapshots';
+
+export class Snapshot {
+  constructor(
+    private model: ResourceSnapshotsModel,
+    private snapshotClient: ResourceSnapshots
```

Comment:
Instead of injecting the client in the ctor, could it be created internally ? 

However, it does ultimately rely on the parent command to make sure the platform-client is authenticated, so.... I think it's fine like this.

Reply:
Without the client passed in the constructor, the `Snapshot` class has to know all the org config variables (org, env and region) and create the appropriate client with that info.
However, I wanted the `SnapshotFactory` class to handle the client generation poutine since the `Snapshot` class does not need to know that information.